modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -45,22 +43,28 @@ def missing_loa
         request.headers['x-va-level-of-assurance'].blank?
       end
 
-      def build_identity
-        UserIdentity.create(uuid: request.headers['x-va-idp-uuid'],
-                            email: request.headers['x-va-user-email'],
-                            first_name: request.headers['x-va-first-name'],
-                            last_name: request.headers['x-va-last-name'],
-                            # TODO: break this out into a method that handles unknown gender
-                            gender: request.headers['x-va-gender']&.chars&.first&.upcase,
-                            birth_date: request.headers['x-va-dob'],
-                            ssn: request.headers['x-va-ssn'],
-                            mhv_icn: request.headers['x-va-mhv-icn'],
-                            dslogon_edipi: request.headers['x-va-dslogon-edipi'],
-                            loa:
-                            {
-                              current: request.headers['x-va-level-of-assurance'].to_i,
-                              highest: request.headers['x-va-level-of-assurance'].to_i
-                            })
+      def save_requested_identity
+        # In addition to constructing the the identity object, we save it in order to prime the
+        # identity cache. This is done because this endpoint is primarily called by the saml-proxy
+        # during the login process. If a user is logging in, usually a client app will make use of
+        # the openid token soon.
+        OpenidUserIdentity.create(
+          uuid: request.headers['x-va-idp-uuid'],
+          email: request.headers['x-va-user-email'],
+          first_name: request.headers['x-va-first-name'],
+          last_name: request.headers['x-va-last-name'],
+          # TODO: break this out into a method that handles unknown gender
+          gender: request.headers['x-va-gender']&.chars&.first&.upcase,
+          birth_date: request.headers['x-va-dob'],
+          ssn: request.headers['x-va-ssn'],
+          mhv_icn: request.headers['x-va-mhv-icn'],
+          dslogon_edipi: request.headers['x-va-dslogon-edipi'],
+          loa:
+          {
+            current: request.headers['x-va-level-of-assurance'].to_i,
+            highest: request.headers['x-va-level-of-assurance'].to_i
+          }
+        )
       end
```

Comment:
should there maybe be an object / class that deserializes, handles, even parses these? It's difficult to get granular unit level testing of each of these attributes being parsed within the controller. It would be a better abstraction to do this in a separate class that handles deserialization.